components/dropdown/dropdown.vue

```diff
@@ -432,7 +432,7 @@ export default {
     },
 
     onKeyValidation (e, eventHandler) {
-      if (this.open !== null && !this.openOnContext) { return; }
+      if (this.open !== null && !this.openOnContext && !this.isOpen) { return; }
```

Comment:
I spoke to Jose about his original intention with this code since it was confusing me. It's basically just to prevent the menu from opening with the arrow keys when it's a context menu. We found a way to greatly simplify this logic since it's very confusing with all these opens and isOpens.

We can just use the `open-with-arrow-keys` prop on popover. Something like this:

```
<dt-popover
 :open-with-arrow-keys="shouldOpenWithArrowKeys"
 ...
...

<script>
...
    shouldOpenWithArrowKeys () {
      return !this.openOnContext;
    },
</script>
```

Then we can probably just get rid of onKeyValidation entirely. Don't think we'll need it anymore.